pkg/components/exporter/exporter.go

```diff
@@ -106,6 +108,74 @@ func NewDaemonSet(detail components.Detail, k *v1alpha1.Kepler) *appsv1.DaemonSe
 
 	bindAddress := "0.0.0.0:" + strconv.Itoa(int(deployment.Port))
 
+	exporterContainer := &corev1.Container{
+		Name:            "kepler-exporter",
+		SecurityContext: &corev1.SecurityContext{Privileged: pointer.Bool(true)},
+		Image:           Config.Image,
+		Command: []string{
+			"/usr/bin/kepler",
+			"-address", bindAddress,
+			"-enable-gpu=$(ENABLE_GPU)",
+			"-enable-cgroup-id=true",
+			"-v=$(KEPLER_LOG_LEVEL)",
+			"-kernel-source-dir=/usr/share/kepler/kernel_sources",
+			"-redfish-cred-file-path=/etc/redfish/redfish.csv",
+		},
+		Ports: []corev1.ContainerPort{{
+			ContainerPort: int32(deployment.Port),
+			Name:          "http",
+		}},
+		LivenessProbe: &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/healthz",
+					Port:   intstr.IntOrString{Type: intstr.Int, IntVal: deployment.Port},
+					Scheme: "HTTP",
+				},
+			},
+			FailureThreshold:    5,
+			InitialDelaySeconds: 10,
+			PeriodSeconds:       60,
+			SuccessThreshold:    1,
+			TimeoutSeconds:      10},
+		Env: []corev1.EnvVar{
+			{Name: "NODE_IP", ValueFrom: k8s.EnvFromField("status.hostIP")},
+			{Name: "NODE_NAME", ValueFrom: k8s.EnvFromField("spec.nodeName")},
+			{Name: "KEPLER_LOG_LEVEL", ValueFrom: k8s.EnvFromConfigMap("KEPLER_LOG_LEVEL", ConfigmapName)},
+			{Name: "ENABLE_GPU", ValueFrom: k8s.EnvFromConfigMap("ENABLE_GPU", ConfigmapName)}},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: "lib-modules", MountPath: "/lib/modules", ReadOnly: true},
+			{Name: "tracing", MountPath: "/sys", ReadOnly: true},
+			{Name: "kernel-src", MountPath: "/usr/src/kernels", ReadOnly: true},
+			{Name: "kernel-debug", MountPath: "/sys/kernel/debug"},
+			{Name: "proc", MountPath: "/proc"},
+			{Name: "cfm", MountPath: "/etc/kepler/kepler.config"},
+		},
+	}
+
+	var containers []corev1.Container
+
+	// exporter volumes
+	var volumes = []corev1.Volume{
+		k8s.VolumeFromHost("lib-modules", "/lib/modules"),
+		k8s.VolumeFromHost("tracing", "/sys"),
+		k8s.VolumeFromHost("proc", "/proc"),
+		k8s.VolumeFromHost("kernel-src", "/usr/src/kernels"),
+		k8s.VolumeFromHost("kernel-debug", "/sys/kernel/debug"),
+		k8s.VolumeFromConfigMap("cfm", ConfigmapName),
+	}
+
+	if estimator.IsEstimatorSidecarEnabled(k) {
```

Comment:
```suggestion
         containers := []corev1.Container{*exporterContainer}
         
	if estimator.NeedsSidecar(k) {
	    containers = addEstimatorSidecar(exporterContainer)
	}
```

Reply:
As it also needs changes on volume, I modify your suggestion as below. 

```go
containers, volumes = addEstimatorSidecar(exporterContainer, volumes)
```